src/Database/Dialect/SqlserverDialectTrait.php

```diff
@@ -325,6 +325,10 @@ public function rollbackSavePointSQL($name)
      */
     public function newCompiler()
     {
+        if (isset($this->_config['compiler'])) {
```

Comment:
This is a fairly low level option. Should we provide any abstractions here? For example

```
'compiler' => 'default | triggers'
```